docs/server-side-rendering.md

```diff
@@ -22,26 +22,11 @@ React components used on the server will be rendered to HTML by being passed to
 
 ### Caching
 
-Because it is necessary to serve the redux state along with a server-rendered page, we use two levels of cache on the server: one to store the redux state, and one to store rendered layouts.
+Because it is necessary to serve the redux state along with a server-rendered page, we use two levels of cache on the server: one to [store the redux state](https://github.com/Automattic/wp-calypso/blob/master/server/state-cache/index.js), and one to [store rendered layouts](https://github.com/Automattic/wp-calypso/blob/7ded1642a5b95a30b64fe0a3e462ddd2317c0df2/server/render/index.js#L33).
 
-##### Data Cache
+Both caches use the same key, which is the pathname of the URL. URLs with query args are not cached unless the arg name is present in `context.queryCacheKeys`, in which case the argument or arguments are appended to the key.
```

Comment:
How about being a bit more specific:

---

[…] unless _all_ of the query arguments are in `queryCacheKeys` […]

If query arguments _not in_ `queryCacheKeys` are detected, server-side rendering is disabled.